community/security/src/main/java/org/neo4j/server/security/auth/BasicAuthSubject.java

```diff
@@ -59,9 +60,21 @@ public AuthenticationResult getAuthenticationResult()
         return authenticationResult;
     }
 
+    /**
+     * Sets a new password for the BasicAuthSubject.
+     *
+     * @param password The new password
+     * @throws IOException If the new user credentials cannot be stored on disk.
+     * @throws IllegalCredentialsException If password is invalid, e.g. if the new password is the same as the current.
+     */
     @Override
-    public void setPassword( String password ) throws IOException
+    public void setPassword( String password ) throws IOException, IllegalCredentialsException
     {
+        if ( user.credentials().matchesPassword( password ) )
+        {
+            throw new IllegalCredentialsException( "Old password and new password cannot be the same." );
+        }
+
```

Comment:
This is kind of a hard-coded rudimentary password policy here, but I think this is fine for BasicAuthSubject. 
Maybe we could break it out to a separate private method in case we also need to check that the password cannot be empty? 
In the old REST UserService the password "policy" is enforced outside the auth manager.